### PR TITLE
Improve the autotailor script

### PR DESCRIPTION
**Pull request by jan-cerny**
This pull request improves the `autotailor` utility. The goal of the improvements is that the `autotailor` can be used for real-world usage and easily extended in future. Specifically, these are the changes made in this PR:
- added unit tests
- added integration test
- refactoring
- coding style
- add options `--rule-role` and `--rule-severity` to refine rules roles and severities

For more details, please read commit messages of every commit.

This PR supersedes PR https://github.com/OpenSCAP/openscap/pull/1877.

### CODE REVIEW

File: tests/utils/autotailor_integration_test.sh [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+assert_exists 1 '/Benchmark/TestResult/rule-result[@idref="xccdf_com.example.www_rule_R4"]/result[text()="notselected"]'
+
+# refine value v1 to 30
+python3 $autotailor --id-namespace "com.example.www" --var-value V1=thirty $ds $original_profile > $tailoring
+$OSCAP xccdf eval --profile P1_customized --progress --tailoring-file $tailoring --results $result $ds
+assert_exists 1 '/Benchmark/TestResult/set-value[@idref="xccdf_com.example.www_value_V1" and text()="thirty"]'
+assert_exists 1 '/Benchmark/TestResult/rule-result[@idref="xccdf_com.example.www_rule_R1"]/result[text()="pass"]'
+assert_exists 1 '/Benchmark/TestResult/rule-result[@idref="xccdf_com.example.www_rule_R2"]/result[text()="pass"]'
+assert_exists 1 '/Benchmark/TestResult/rule-result[@idref="xccdf_com.example.www_rule_R3"]/result[text()="notselected"]'
+assert_exists 1 '/Benchmark/TestResult/rule-result[@idref="xccdf_com.example.www_rule_R4"]/result[text()="notselected"]'
```
Comment (evgenyz) [inline tests/utils/autotailor_integration_test.sh:103]: No newline.

File: utils/autotailor.8 [hunk 2]
```diff
@@ -5,7 +5,7 @@ autotailor \- CLI tool for tailoring of SCAP data streams.
 autotailor produces tailoring files that SCAP-compliant scanners can use to complement SCAP data streams.
 A tailoring file adds a new profile, which is supposed to extend a profile that is already present in the data stream.
 
-Tailoring can add or remove rules, and it can redefine contents of XCCDF variables.
+Tailoring can add or remove rules, refine rules, and it can redefine contents of XCCDF variables.
```
Comment (evgenyz) [inline utils/autotailor.8:8]:
```suggestion
Tailoring can add, remove or refine rules, and it also can redefine contents of XCCDF variables.
```

**Review by evgenyz — CHANGES_REQUESTED** [review]
In general I approve the changes. Awesome! Thank you.

There are a couple of nitpicks, though. See individual notes.

Also I imagined that a new tool could use little bit more of modern syntax sugar. What do you think, should we embrace f-strings in autotailor?

File: utils/autotailor [hunk 3]
```diff
@@ -26,37 +27,121 @@ import xml.etree.ElementTree as ET
...
+            msg = (
+                "Unsupported refine-rule attribute {attribute}".format(
+                    attribute=attribute))
+            raise ValueError(msg)
+        return enumeration
+
+    @staticmethod
+    def _validate_rule_refinement_params(rule_id, attribute, value):
+        if not is_valid_xccdf_id(rule_id):
+            msg = "Rule id '{rule_id}' is invalid!".format(rule_id=rule_id)
```
Comment (evgenyz) [inline utils/autotailor:93]: As we operate with Python 3 in 1.3.x we probably can do this:

```suggestion
            msg = f"Rule id '{rule_id}' is invalid!"
```

File: utils/autotailor [hunk 4]
```diff
@@ -71,45 +156,58 @@ class Tailoring:
...
 
-        benchmark = ET.SubElement(root, "xccdf-1.2:benchmark")
-        datastream_uri = pathlib.Path(self.original_ds_filename).absolute().as_uri()
+        benchmark = ET.SubElement(root, "{%s}benchmark" % NS)
+        datastream_uri = pathlib.Path(
+            self.original_ds_filename).absolute().as_uri()
         benchmark.set("href", datastream_uri)
 
-        version = ET.SubElement(root, "xccdf-1.2:version")
+        version = ET.SubElement(root, "{%s}version" % NS)
```
Comment (evgenyz) [inline utils/autotailor:192]: And this:
```suggestion
        version = ET.SubElement(root, f"{NS}version")
```

File: utils/autotailor [hunk 5]
```diff
@@ -154,14 +272,15 @@ def parse_args():
...
         "-p", "--new-profile-id",
         help="Specify the ID of the tailored profile. "
-        "The ID of the new profile can be either its full ID, or the suffix, in which case "
-        "the 'xccdf_<id-namespace>_profile_' prefix will be prepended internally. "
-        "If left out, the new ID will be obtained "
+        "The ID of the new profile can be either its full ID, or the suffix, "
+        "in which case the 'xccdf_<id-namespace>_profile_' prefix will be "
+        "prepended internally. If left out, the new ID will be obtained "
         "by appending '{suffix}' to the tailored profile ID."
         .format(suffix=DEFAULT_PROFILE_SUFFIX))
```
Comment (evgenyz) [inline utils/autotailor:279]: But _this_ might not be worthy of converting into a block of f-strings.

File: utils/autotailor [hunk 6]
```diff
@@ -26,37 +27,121 @@ import xml.etree.ElementTree as ET
...
+            raise ValueError(msg)
+        enumeration = Tailoring._find_enumeration(attribute)
+        if value in enumeration:
+            return
+        allowed = ", ".join(map(quote, enumeration))
+        msg = (
+            "Can't refine {attribute} of rule '{rule_id}' to '{value}'. "
+            "Allowed {attribute} values are: {allowed}.".format(
+                attribute=attribute, rule_id=rule_id, value=value,
+                allowed=allowed))
```
Comment (evgenyz) [inline utils/autotailor:103]: And this: `f"Can't refine {attribute} of rule '{rule_id}' to '{value}'. Allowed {attribute} values are: {allowed}."`.

File: utils/autotailor (hunk 4 above)
Comment (jan-cerny) [reply utils/autotailor:192]: You actually need to escape the braces: `f"{{{NS}}}version"`, so here I prefer to use the `%`  form.

Comment (evgenyz) [reply utils/autotailor:184]: Right.

**Pull request merged**